CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs

```diff
@@ -0,0 +1,293 @@
+            //Act
+            var result = await _homeworkService.CreateHomeworkAsync(homeworkRequestDto);
+
+            //Assert
+            result.Data.Should().NotBeNull();
+            result.Data.Should().BeEquivalentTo(_mapper.Map<HomeworkDto>(homeworkDto));
+        }
+        #endregion
+
+        #region UpdateHomeworkAsync
+
+        [Fact]
+        public async Task UpdateHomeworkAsync_DefaultHomework_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = default;
+
+            //Act
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+
+        }
+
+        [Fact]
+        public async Task UpdateHomeworkAsync_NotExistingHomeworkId_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z")
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(true);
+            _homeworkRepositoryMock.Setup(x => x.GetByIdAsync(It.IsAny<long>())).ReturnsAsync(default(Homework));
+
+            _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            //Act
+
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+
+        }
+
+        [Fact]
+        public async Task UpdateHomeworkAsync_ValidData_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z")
+            };
+
+            Homework homework = new Homework
+            {
+                Id = 1,
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z"),
+
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(true);
+            _homeworkRepositoryMock.Setup(x => x.GetByIdAsync(It.IsAny<long>())).ReturnsAsync(homework);
+
+            _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            //Act
+
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Data.Should().BeEquivalentTo(new HomeworkDto
```

Comment:
Can we reuse `homework` variable from line 193?

Reply:
Done